tracee-ebpf/tracee/argprinters.go

```diff
@@ -51,130 +51,212 @@ func (t *Tracee) parseArgs(event *external.Event) error {
 		if int32(event.EventID) == CapCapableEventID {
 			if capArg := getEventArg(event, "cap"); capArg != nil {
 				if capability, isInt32 := capArg.Value.(int32); isInt32 {
-					capArg.Value = helpers.ParseCapability(capability)
 					capArg.Type = "string"
+					capArg.Value = "" // arg has to always exist (even if errors below)
+					capabilityFlagArgument, err := helpers.ParseCapability(uint64(capability))
+					if err != nil {
+						return err
```

Comment:
returning an error will skip this event - I don't think this is the behavior we want

Reply:
Ok, based on your feedback:

1) @grantseltzer what was the reasoning behind this change ?

2) Should we revert the libbpfgo change and keep things as they were ?

3) I agree about the error handling. The fact that some argument might not be correctly parsed doesn't necessarily mean we should skip the entire event.
  - I wasn't sure about returning errors but, since the new API had them, I thought it was because you wanted this behavior (assumed this was discussed at somepoint).

Reply:
Let's see how the parser wrapper idea fits into this and then we see the overall change again.

Reply:
Changes were for reducing code duplication, surfacing errors, and so we can define constants to avoid having redefined strings everywhere.